server/src/db.rs

```diff
@@ -232,6 +234,33 @@ impl Db {
     pub fn next_sequence(&self) -> u64 {
         self.sequence.fetch_add(1, Ordering::SeqCst)
     }
+
+    /// Drops partitions from the mutable buffer if it is over size
+    pub fn check_size_and_drop_partitions(&self) -> Result<()> {
+        if let (Some(db), Some(config)) = (&self.mutable_buffer, &self.rules.mutable_buffer_config)
+        {
+            let mut size = db.size();
+            if size > config.buffer_size {
+                let mut partitions = db.partitions_sorted_by(&config.partition_drop_order);
+                while let Some(p) = partitions.pop() {
+                    let p = p.read().expect("mutex poisoned");
+                    let partition_size = p.size();
+                    size -= partition_size;
+                    let key = p.key();
+                    db.drop_partition(key);
+                    info!(
+                        "dropped partition {} from mutable buffer to free {} bytes",
+                        key, partition_size
+                    );
```

Comment:
It might be cool to use structured logging in this case
```suggestion
                    info!(
                        "dropped partition from mutable buffer",
                        partition_key=key, partition_size
                    );
```
So that `partition_key` and `partition_size` show up as fields in the logs rather than embedded in the message

Reply:
I'm going to make this change. The difference in the log output is:

```diff
- Feb 22 12:40:44.002  INFO server::db: dropped partition p1 from mutable buffer to free 131 bytes
+ Feb 22 12:44:45.116  INFO server::db: dropped partition from mutable buffer partition_key="p1" partition_size=131
```